### ISSUE TITLE
Proxy: ToPropertyDescriptor assert; when Proxy has() always lies and returns true, should result in TypeError

### ISSUE THREAD

**Issue by dilijev** [Microsoft/ChakraCore#884]
Repro test case:

```js
var h = { has : function() { WScript.Echo("has"); return true; }};
var p = new Proxy({}, h);
var o = {};
Object.defineProperty(o, "x", p);
```
 
Output:
```
has
ASSERTION 3060: (D:\src\Chakra\core\lib\Runtime\Language\JavascriptOperators.cpp, line 8661) Proxy : HasProperty and GetProperty's result don't match for 'enumerable'.  Failure: (0)
FATAL ERROR: jshost.exe failed due to exception code c0000420
```
 
Expected output:
```
has
has
has
has
has
has
TypeError: Property cannot have both accessors and a value
```

Each `has` is for the Proxy properties: configurable, enumerable, value, writable, get, and set that are checked when there is a proxy around the PropertyDescriptor being used with Object.defineProperty.

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy
See: https://tc39.github.io/ecma262/2016/#sec-object.defineproperty
See: https://tc39.github.io/ecma262/2016/#sec-topropertydescriptor

The reason there should be a `TypeError` is because according to spec semantics, when the proxy says `has()` is true for a property, we should try to look it up, and if we discover that it is not present, then we should set it to the default value (falsy: undefined for functions and false for non-functions). Since certain combinations of these PropertyDescriptor properties are disallowed, we see a `TypeError` in this case.

Converted from [7019635](https://microsoft.visualstudio.com/web/wi.aspx?pcguid=cb55739e-4afe-46a3-970f-1b49d8ee7564&id=7019635)

/cc @bterlson to double check details.

**Comment by abchatra** [comment]
@dilijev if the issue is fixed, can this be closed?

**Comment by dilijev** [comment]
@abchatra fix is not merged in yet - see #885
